src/cmsg.rs

```diff
@@ -0,0 +1,365 @@
+use crate::sys;
+use std::borrow::Borrow;
+use std::convert::TryInto as _;
+use std::io::IoSlice;
+use std::iter::FromIterator;
+
+#[derive(Debug, Clone)]
+struct MsgHdrWalker<B> {
```

Comment:
Let's remove the generic parameter, we only need `&[u8]`, same for the other structures.

Reply:
Tentatively keeping this parameter so we can reuse some logic between `&[u8]` and `&mut [u8]`, but removed all the type parameters from the public API for now.